api/handlers_instances.go

```diff
@@ -516,6 +516,9 @@ func (s *server) InstanceUpdateHandler(w http.ResponseWriter, r *http.Request) {
 	} else if len(req.Tags) > 0 && len(req.InstanceType) > 0 {
 		handleError(w, apierror.New(apierror.ErrBadRequest, "only one of these fields should be provided: tags or instance_type", nil))
 		return
+	} else if _, ok := req.InstanceType["value"]; len(req.InstanceType) > 0 && !ok {
```

Comment:
Why don't we move this check under `else if len(req.InstanceType) > 0 {` below? This seems convoluted.